src/content/blog/ddev-backups.md

```diff
@@ -0,0 +1,55 @@
+
+**But try not to rely on the local computer anyway.** I try to push code and config to my Git provider (GitHub, GitLab, etc.) and make sure I know how to recreate any user-generated files. Make sure your database contents are accessible from upstream sources like production, or keep good snapshots or exports.
+
+**Databases** need an extra step for safety. In general, don't do work on a database that cannot be recreated by code (with migrations, for example). But most of us want to have a database quickly available as a good place to start from. Since databases are typically in a binary format that can't reliably be backed up, we need a good way to make a copy. DDEV has two great ways to turn databases into files, `ddev snapshot` and `ddev export-db`.
+
+* **[`ddev snapshot`](https://ddev.readthedocs.io/en/stable/users/usage/cli/#snapshotting-and-restoring-a-database)**, optionally with a `name` argument, takes a binary copy of all of your databases (most projects have just one) and saves it into the `.ddev/db_snapshots` directory as a gzipped binary file. You can snapshot all your registered projects with `ddev snapshot -a`.
+
+* **[`ddev export-db`](https://ddev.readthedocs.io/en/stable/users/usage/commands/#export-db)** exports a text-based dump of a single database to a named file. For example, `ddev export-db --file=.tarballs/db.sql.gz`. A text-based file can be imported into databases of related types, and not just its source type. For example, a MariaDB 10.11 database dump in this format can typically be imported into a MySQL 8.0 database.
+
+* **Automating database backups**: Some people like to use a pre-stop hook to do a database backup, so that a database dump happens every time they do `ddev stop` or `ddev poweroff`. For example:
+
+```yaml
+hooks:
+  pre-stop:
+    - exec-host: "ddev snapshot"
+```
+
+If you prefer a text-based database backup, you could do this:
+```yaml
+hooks:
+  pre-stop:
+    - exec-host: 'mkdir -p .tarballs && ddev export-db --file=.tarballs/db.$(date +"%Y%m%d%H%M%S").sql.gz'
+```
+
+**PHP Code and Configuration** are text files that should be under control of your Git environment. The easy answer for those is to always keep them pushed up to your Git provider. Work on a branch and a new commit up to your branch regularly, so you could start afresh on any computer any time. This is a wonderful thing about source control!
+
+**User-generated and non-code images, etc.**: In general, these should be coming from upstream hosting environments. Although it is possible to push them upstream, it's usually a bad idea, as it could overwrite a production environment. So normally if you lost these, you would just pull them again from wherever they come from.
```

Comment:
```suggestion
**User-generated and non-code images, etc.**: In general, these should come from upstream hosting environments. Although it is possible to push them upstream, it's usually a bad idea, as this could overwrite a production environment. So normally, if you lose these, you would just pull them again from wherever they come from.
```